src/platform/utilities/scroll/scroll.js

```diff
@@ -75,103 +75,112 @@ export const scrollToTop = async (
-      ).some(({ shadowRoot }) =>
-        shadowRoot?.querySelector('va-modal[visible]:not([visible="false"])'),
+  const { focusOnAlertRole = false, errorContext } = options;
+  const selectors = ERROR_ELEMENTS.join(',');
+  const el = document.querySelector(selectors);
+
+  if (!el) {
+    console.warn('scrollToFirstError: Error element not found');
+    if (
+      !environment.isProduction() &&
+      Array.isArray(errorContext) &&
+      errorContext.length &&
+      errorContext.some(err =>
+        err?.stack?.includes('You must provide a response'),
+      )
+    ) {
+      const fieldKeys = errorContext
+        .filter(err => err?.stack?.includes('You must provide a response'))
+        .map(err => err.stack.split(':')[0]);
+      console.warn(
+        `Schema validation error: The following fields are marked as required in your schema but do not exist in the page:\n${fieldKeys
+          .map(key => `- "${key}"`)
+          .join(
+            '\n',
+          )}\nCheck your required fields in the schema match the fields in your UI.`,
       );
-      const isModalOpen =
-        document.body.classList.contains('modal-open') ||
-        document.querySelector('va-modal[visible]:not([visible="false"])') ||
-        isShadowRootModalOpen;
-
-      // prevent page scroll if there is an open modal, as the error could
-      // be within the modal itself
-      if (!isModalOpen) {
-        const position = getElementPosition(el);
-        scrollTo(position - 10, options);
-
-        if (focusOnAlertRole) {
-          // Adding a delay so that the shadow DOM needs to render and attach
-          // before we try to focus on the element; without the setTimeout,
-          // focus ends up staying on the "Continue" button
-          requestAnimationFrame(() => {
-            focusElement('[role="alert"]', {}, el?.shadowRoot);
-          });
-        } else {
-          focusElement(el);
-        }
-      }
+    }
+    return;
+  }
+
+  // check for any modals that would interfere with scrolling
+  const isShadowRootModalOpen = Array.from(
+    document.querySelectorAll('va-omb-info'),
+  ).some(({ shadowRoot }) =>
+    shadowRoot?.querySelector('va-modal[visible]:not([visible="false"])'),
+  );
+  const isModalOpen =
+    document.body.classList.contains('modal-open') ||
+    document.querySelector('va-modal[visible]:not([visible="false"])') ||
+    isShadowRootModalOpen;
+
+  // prevent page scroll if there is an open modal, as the error could
+  // be within the modal itself
+  if (!isModalOpen) {
+    const position = getElementPosition(el);
+    await scrollTo(position - 10, options);
+
+    if (focusOnAlertRole) {
+      // Set up aria-describedby for all elements in error state
+      const allErrorElements = document.querySelectorAll(selectors);
+      allErrorElements.forEach(errorWebComponent => {
+        const errorElement = errorWebComponent?.shadowRoot?.querySelector(
+          '[role="alert"], #input-error-message',
```

Comment:
> [!note]
> This check feels fragile. At the design system level, we should standardize the id names for our error messages so that we can more reliably target them.